lib/id3c/cli/command/etl/__init__.py

```diff
@@ -123,7 +123,8 @@ def upsert_encounter(db: DatabaseSession,
     with db.cursor() as cursor:
         cursor.execute("""
             select encounter_id as id,
-                identifier
+                identifier,
+                md5(row (encounter.individual_id, encounter.site_id, encounter.encountered, encounter.age, encounter.details)::text) as row_hash
```

Comment:
I like this hashing approach overall, and agree it should reduce updates in the table. If we added more columns to the table, I imagine we'd have to update here as well? 

I'm assuming you didn't include all columns in the hash because you wanted to exclude created/modified columns. I wonder how often we'd get "updates" due to a difference in just those columns if all columns were included in the hash. From a maintenance perspective, it might be preferable to include all the columns, but I'm not sure how much the tradeoff would end up being.

Reply:
Right, I only included the columns that the `upsert_encounter` function it attempting to set. The created and modified dates don't need to be considered when determining if the data has changed. If we add more columns that would be set via `upsert_encounter` then yes, they would need to be added to the hash.

Reply:
Understood about created/modified not *needing* to be considered. I do still have some concerns that this creates non-obvious implications for schema changes down the line and makes the code more fragile. If this is currently the only place programatically modifying encounters, there should only be nominal functional difference in the amount of updates if it includes all columns. 
As an alternative to including all columns in the hash, we could explicilty ignore the created/modified columns with an approach like https://blog.jooq.org/selecting-all-columns-except-one-in-postgresql/

Reply:
I see what you're getting at, but any schema changes would have already required multiple changes to `upsert_encounter`, including the parameters and `data` dict. Agree that updating the hashed columns would be less obvious, but the benefits to hardcoding them this way is that the incoming `data` values can be cast to match how they would appear if they were already in the table, so when hashed they will match the value coming from the `select` query. The hash values are different without that casting.

If you see a way that can work, let me know. Otherwise I can add some comments to the code explaining the need to update those lines if there are schema changes.

Reply:
Ok! Thanks for the discussion here Dave. I think adding a comment would be helpful. I think thats all of the feedback I have, so I'll go ahead and mark approved from my end.